File: Rust/LinkedList/py_linked_list/node.py

class Node(object):
    def __init__(self, _elem=None, _next=None):
        self._elem = _elem
        self._next = _next

    def __str__(self):
        return str(self._elem)

    # __repr__ = __str__
    
    def value(self):
        return self._elem

    def next(self):
        return self._next

    def setValue(self, e):
        self._elem = e

    def setNext(self, n):
        self._next = n


if __name__ == '__main__':
    n = Node('abc', None)
    n.setNext(Node(3, None))
    print n
    print n.next()

File: Rust/LinkedList/LinkedListScala/Node.scala
package LinkedListScala {

class Node[T](var e: T = T(0), var n: Node[T] = null, var a: Node[T] = null) {

  protected _elem: T = e;
  protected _next: Node[T] = n;
  protected _arb: Node[T] = a;
}
}
